gufe/base.py

```diff
@@ -0,0 +1,108 @@
+import abc
+import uuid
+import sys
+import threading
+import hashlib
+import inspect
+from packaging.version import parse as parse_version
+from typing import Dict, Any, Callable
+
+
+class GufeTokenizableMixin:
+    """Mixin for all tokenizble tokenizeable gufe objects.
+
+    """
+
+    @property
+    def token(self):
+        if not hasattr(self, '_token') or self._token is None:
+            self._token = tokenize(self)
+        return self._token
+
+    @property
+    def key(self):
+        if not hasattr(self, '_key') or self._key is None:
```

Comment:
I don't see any usage of `_key`?  Why not just have one way of doing this

Reply:
We might just want to drop the `token` property, since we use `key` for stubbing (keyencode/decode).

Reply:
Removed `token` property to reduce the number of items in play. We just use `key`.